privacyidea/lib/utils.py

```diff
@@ -398,19 +401,14 @@ def parse_date(date_string):
             td = timedelta(days=delta_amount)
         else:
             td = timedelta()
-        return datetime.now() + td
+        return datetime.now(tzlocal()) + td
 
     # check 2016/12/23, 23.12.2016 and including hour and minutes.
     d = None
-    for date_format in ["%Y/%m/%d", "%d.%m.%Y",
-                        "%Y/%m/%d %I:%M%p",
-                        "%d.%m.%Y %H:%M"]:
-        try:
-            d = datetime.strptime(date_string, date_format)
-            break
-        except ValueError:
-            log.debug("Dateformat {1!s} did not match date {0!s}".format(
-                date_string, date_format))
+    try:
+        d = parse_date_string(date_string)
```

Comment:
Please see 83769a6dae12f1381be3b1b9f3329431abbd4c42 
I think this is not the most sensible and most definite way?